src/types/parity_pending_transaction.rs

```diff
@@ -0,0 +1,135 @@
+use serde::{
+    ser::{SerializeMap, Serializer},
+    Serialize,
+};
+
+use super::{Address, U256, U64};
+
+/// Condition to filter pending transactions
+#[derive(Clone)]
+pub enum FilterCondition<T> {
+    /// Lower Than
+    Lt(T),
+    /// Equal
+    Eq(T),
+    /// Greater Than
+    Gt(T),
+}
+
+/// To Filter
+#[derive(Clone)]
+pub enum ToFilter {
+    /// Address
+    Address(Address),
+    /// Action (i.e. contract creation)
+    Action,
+}
+
+/// Filter for pending transactions (only openethereum/Parity)
+#[derive(Clone, Default, Serialize)]
+pub struct ParityPendingTransactionFilter {
+    /// From address
+    #[serde(skip_serializing_if = "Option::is_none")]
+    pub from: Option<FilterCondition<Address>>,
+    /// To address or action, i.e. contract creation
+    #[serde(skip_serializing_if = "Option::is_none")]
+    pub to: Option<ToFilter>,
+    /// Gas
+    #[serde(skip_serializing_if = "Option::is_none")]
+    pub gas: Option<FilterCondition<U64>>,
+    /// Gas Price
+    #[serde(skip_serializing_if = "Option::is_none")]
+    pub gas_price: Option<FilterCondition<U64>>,
+    /// Value
+    #[serde(skip_serializing_if = "Option::is_none")]
+    pub value: Option<FilterCondition<U256>>,
+    /// Nonce
+    #[serde(skip_serializing_if = "Option::is_none")]
+    pub nonce: Option<FilterCondition<U256>>,
+}
+
+impl<T> Serialize for FilterCondition<T>
+where
+    T: Serialize,
+{
+    fn serialize<S>(&self, serializer: S) -> Result<S::Ok, S::Error>
+    where
+        S: Serializer,
+    {
+        let mut map = serializer.serialize_map(Some(1))?;
+        match self {
+            Self::Lt(v) => map.serialize_entry("lt", v),
```

Comment:
I'm pretty sure it's totally possible to do with `serde` attributes instead of writing a custom serialize implementation.